src/babel-plugin-dynamic-import-amd.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import dyanamicImportSyntax from '@babel/plugin-syntax-dynamic-import';
+import {NodePath} from '@babel/traverse';
+import {CallExpression, Identifier, Program, Statement} from 'babel-types';
+import template from '@babel/template';
+
+const ast = template.ast;
+
+export const dynamicImportAMD = {
+  inherits: dyanamicImportSyntax,
+
+  visitor: {
+    Program(path: NodePath<Program>) {
+      const identifiers = new Set<string>();
+      const dynamicImports: NodePath<CallExpression>[] = [];
+      path.traverse({
+        CallExpression(path: NodePath<CallExpression>) {
+          if (path.node.callee.type as string === 'Import') {
+            dynamicImports.push(path);
+            const bindings = path.scope.getAllBindings();
+            for (const name of Object.keys(bindings)) {
+              identifiers.add(name);
+            }
+          }
+        }
+      });
+      if (dynamicImports.length === 0) {
+        return;
+      }
+      let requireId: Identifier|undefined = undefined;
+      do {
+        requireId = path.scope.generateUidIdentifier('require');
+      } while (identifiers.has(requireId.name));
+      const statements = path.node.body as Statement[];
+      statements.unshift(ast`import * as ${requireId} from 'require';`);
+      for (const importPath of dynamicImports) {
+        const specifier = importPath.node.arguments[0];
+        // The 2-argument local require takes an array of module IDs and
```

Comment:
This comment seems more relevant just above, before you add the require import.

Reply:
updated comments